EPI/parse-nested-GrFN.py

```diff
@@ -0,0 +1,112 @@
+import json
+from argparse import ArgumentParser
+
+
+def formatGraph(data): 
+    modelMetadata = {}
+    nodesDict = {}
+    nodes = []
+    edges = []
+
+    # FIXME: We might want to only check for fields we are interested in instead of just going over all items. 
+    for key,value in data.items():
+        if key == 'variables':
+            for variable in value: 
+                metadata = variable['metadata'][0] if 'metadata' in variable else {}
+
+                variable['nodeType'] = 'variable'
+                variable['metadata'] = metadata
+                nodesDict[variable['uid']] = variable
+        if key == 'functions':
+            for function in value: 
+                metadata = function['metadata'][0] if 'metadata' in function else {}
+ 
+                function['nodeType'] = 'function'
+                function['metadata'] = metadata
+                nodesDict[function['uid']] = function
+        if key == 'hyper_edges':
+            for edge in value: 
+                [edges.append({ 'source': i, 'target': edge['function']}) for i in edge['inputs']]
+                [edges.append({ 'source': edge['function'], 'target': o }) for o in edge['outputs']]
+            for i in range(len(edges)):
+                edges[i]['id'] = i
+
+        if key == 'subgraphs':
+            for subgraph in value: 
+                # Get parent name
+                parent_name = subgraph['scope']
+                if (parent_name == '@global'):
+                    parent_name = 'root'
+
+                # Get container name
+                splitted_id = subgraph['basename'].split('.')
+
+                #Get container metadata
+                metadata = subgraph['metadata'][0] if 'metadata' in function else {}
+
+                node = { 'id': subgraph['basename'], 'concept': splitted_id[(len(splitted_id) - 1)], 'nodeType': subgraph['type'], 'label': splitted_id[(len(splitted_id) - 1)], 'parent': parent_name, 'metadata': metadata }
+                nodes.append(node)
+                for n in subgraph['nodes']:
+                    found = nodesDict[n]
```

Comment:
I think if `n` is not in `nodesDict` then a key error will be raised. I think we have to do something like:
```
found = n in nodesDict
if found:
  found_node = nodesDict[n]
  ...
else:
  raise Exception(n + ' Node missing')
```

Reply:
Done.